security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysis.java

```diff
@@ -1,74 +1,207 @@
+                                                                                List<SecurityAnalysisInterceptor> interceptors) {
+            Objects.requireNonNull(network, "Network should not be null");
+            Objects.requireNonNull(workingStateId, "WorkingVariantId should not be null");
+            Objects.requireNonNull(detector, "LimitViolation detector should not be null");
+            Objects.requireNonNull(filter, "LimitViolation filter should not be null");
+            Objects.requireNonNull(computationManager, "ComputationManager should not be null");
+            Objects.requireNonNull(contingenciesProvider, "Contingencies provider should not be null");
+            Objects.requireNonNull(parameters, "Sensitivity analysis parameters should not be null");
+            Objects.requireNonNull(interceptors, "Interceptor list should not be null");
+            return provider.runWithLog(network, workingStateId, detector, filter, computationManager, parameters, contingenciesProvider, interceptors);
+        }
+
+        public SecurityAnalysisResult run(Network network,
+                                          String workingStateId,
+                                          LimitViolationDetector detector,
+                                          LimitViolationFilter filter,
+                                          ComputationManager computationManager,
+                                          SecurityAnalysisParameters parameters,
+                                          ContingenciesProvider contingenciesProvider,
+                                          List<SecurityAnalysisInterceptor> interceptors) {
+            return runAsync(network, workingStateId, detector, filter, computationManager, parameters, contingenciesProvider, interceptors).join();
+        }
+
+        public SecurityAnalysisResult run(Network network, LimitViolationFilter filter,
+                                          ComputationManager computationManager) {
+            return run(network, network.getVariantManager().getWorkingVariantId(), new DefaultLimitViolationDetector(), filter, computationManager, SecurityAnalysisParameters.load(), new EmptyContingencyListProvider(), Collections.emptyList());
+        }
+
+        public SecurityAnalysisResult run(Network network, ComputationManager computationManager) {
+            return run(network, LimitViolationFilter.load(), computationManager);
+        }
+
+        public SecurityAnalysisResult run(Network network) {
+            return run(network, LocalComputationManager.getDefault());
+        }
+
+        @Override
+        public String getName() {
+            return provider.getName();
+        }
+
+        @Override
+        public String getVersion() {
+            return provider.getVersion();
+        }
+    }
 
     /**
-     * To be consistent with {@link #run(String, SecurityAnalysisParameters, ContingenciesProvider)}, this method would also complete exceptionally
```

Comment:
this delete javadoc is no longer relevant ? Or should it be restored ? With the code example showing the typical exception handling

Reply:
Comments was restored intto SecurityAnalysisProvider class